File: 2019-12-5/TestViewWillAppear/TestViewWillAppear/ViewController.m
//
//  ViewController.m
//  TestViewWillAppear
//
//  Created by 吉腾蛟 on 2019/12/13.
//  Copyright © 2019 吉腾蛟. All rights reserved.
//

#import "ViewController.h"
#import "SecondViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)pushAction:(UIButton *)sender {
    SecondViewController *vc=[[SecondViewController alloc] init];
//    vc.modalPresentationStyle=UIModalPresentationFullScreen;
    [self presentViewController:vc animated:true completion:nil];
}
- (void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    
    NSLog(@"调用了");
}

@end

File: 2019-5-29(SWift)/Buttn Fun/Buttn Fun/ViewController.swift
//
//  ViewController.swift
//  Buttn Fun
//
//  Created by admin on 2019/5/29.
//  Copyright © 2019 JY. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var statusLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }


    @IBAction func buttonPressed(_ sender: UIButton) {
        
        let title = sender.title(for: .normal)
        let plainText = "\(String(describing: title)) button pressed"
//        statusLabel.text=plainText
        
        let styledText = NSMutableAttributedString(string: plainText)
        
        let attributes = [NSAttributedString.Key.font: UIFont.boldSystemFont(ofSize: statusLabel.font.pointSize)]
        
        let nameRange = (plainText as NSString).range(of: title!)
        
        styledText.setAttributes(attributes, range: nameRange)
        
        statusLabel.attributedText = styledText
    }
}

